UPGRADE-3.4.md

```diff
@@ -4,6 +4,10 @@ UPGRADE FROM 3.3 to 3.4
 DependencyInjection
 -------------------
 
+ * Autowiring service auto-registration is deprecated and won't be supported
+   in Symfony 4.0. Explicitely inject your dependencies or create services
+   whose ids are their fully-qualified class name.
```

Comment:
Indeed, that's a deprecation that is hard to illustrate but I added an example that I tried to make as familiar as possible :)